WalletWasabi.Daemon/WasabiAppBuilder.cs

```diff
@@ -0,0 +1,179 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using NBitcoin;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+using WalletWasabi.Services;
+using WalletWasabi.Services.Terminate;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Daemon;
+
+public class WasabiApplication
+{
+	public WasabiAppBuilder AppConfig { get; }
+	public Global? Global { get; private set; }
+	public Config Config { get; }
+	public SingleInstanceChecker SingleInstanceChecker { get; }
+
+	public WasabiApplication(WasabiAppBuilder wasabiAppBuilder)
+	{
+		AppConfig = wasabiAppBuilder;
+		Config = new Config(LoadOrCreateConfigs(), wasabiAppBuilder.Arguments);
+		SetupLogger();
+		Logger.LogDebug($"Wasabi was started with these argument(s): {(AppConfig.Arguments.Any() ? string.Join(" ", AppConfig.Arguments) : "none")}.");
```

Comment:
nit: `DefaultIfEmpty` instead of `Any` and the condition. Under the hood `IEnumerable<string?>` overload uses a stack allocated string builder with a 256 character long buffer, so even if performance doesn't matter as in that case I would go with LINQ.